components/proto/src/lib.rs

```diff
@@ -39,6 +39,7 @@
 #[macro_use]
 extern crate serde_derive; // Required for Protobuf.
 
+pub use failure;
```

Comment:
This isn't what "reexporting necessary types" means. Consider `exonum-merkledb`. Here are its reexports:

https://github.com/exonum/exonum/blob/869c61f42b47a6b56a44232f6dc05fdfab6c6e88/components/merkledb/src/lib.rs#L168-L173

...and here's the code that uses them in `exonum-derive`:

https://github.com/exonum/exonum/blob/869c61f42b47a6b56a44232f6dc05fdfab6c6e88/components/derive/src/db_traits.rs#L88-L93

Like in the case of `exonum_merkledb`, you need to create reexports in a hidden module (meaning that the existence of a module is an implementation detail) and use them in the derive macro.